File: app/src/main/java/com/antyzero/smoksmog/ui/screen/start/item/ViewDelegate.java
package com.antyzero.smoksmog.ui.screen.start.item;

import android.view.ViewGroup;

public abstract class ViewDelegate<T extends ListViewHolder<R>, R> {

    private final int viewType;

    public ViewDelegate( int viewType ) {
        this.viewType = viewType;
    }

    public int getViewType() {
        return viewType;
    }

    public abstract T onCreateViewHolder( ViewGroup parent );

    public void onBindViewHolder( T holder, R data ) {
        holder.bind( data );
    }
}

File: app/src/main/kotlin/com/antyzero/smoksmog/ui/screen/start/item/ViewDelegate.kt
package com.antyzero.smoksmog.ui.screen.start.item

import android.view.ViewGroup

abstract class ViewDelegate<T : ListViewHolder<R>, R>(val viewType: Int) {

    abstract fun onCreateViewHolder(parent: ViewGroup): T

    fun onBindViewHolder(holder: T, data: R) {
        holder.bind(data)
    }
}
